.github/workflows/deploy-env.yaml

```diff
@@ -0,0 +1,52 @@
+name: Deploy Environment
+
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        type: choice
+        required: true
+        default: 'ops-preview-1'
+        options:
+        - ALL
+        - LIVE
+        - live-sanchonet@eu-central-1@v2
+        - DEV
+        - live-mainnet@eu-central-1@v2
+        - live-sanchonet@us-east-2@v1
+        - dev-mainnet@us-east-1
+        - live-mainnet@us-east-2@v2
+        - live-sanchonet@us-east-2@v2
+        - dev-preprod@us-east-1@v2
+        - live-preprod@eu-central-1@v2
+        - OPS
+        - dev-preview@us-east-1
+        - live-preprod@us-east-2@v2
+        - ops-preprod-1@us-east-1
+        - dev-sanchonet@us-east-1@v1
+        - live-preview@eu-central-1@v2
+        - ops-preview-1@us-east-1
+        - dev-sanchonet@us-east-1@v2
+        - live-preview@us-east-2@v2
+        - staging-preprod@us-east-1@v2
+
+jobs:
+  deploy:
+    environment: ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: 📥 Checkout repository
+        uses: actions/checkout@v3
+
+      - name: 🧰 Setup Nix
+        uses: cachix/install-nix-action@v21
+
+      - name: 🚀 Deploy
```

Comment:
Wait, but don't we have to build the images first? 🤔 Or at least make sure they're built and pushed to ECR?

Reply:
I thought about it, it should not have `push` at the top and then it should be fine. We may encounter a race condition once in a while but that's probably not going to be even noticable

Reply:
Yes, I plan to take out `push:` once the PR is tested and cleared for merging.

Ref: https://github.com/input-output-hk/cardano-js-sdk/pull/1382#issuecomment-2249679228

Reply:
Okay, thanks. I remember I [added a check in dapp-store once](https://github.com/input-output-hk/dapps/blob/66875ff0e5a755b5477d913b2224b944620a6af2/.github/workflows/ci-cd.yml#L44-L54), to prevent such a race condition, but they use Git commits to tag images.